src/index.ts

```diff
@@ -231,31 +229,43 @@ export const handleRotateKey = async (ctx: Context, _request?: Request) => {
 
 const handleClearKey = async (ctx: Context, _request?: Request) => {
 	ctx.metrics.keyClearTotal.inc();
-
-	const now = Date.now();
+	const now = new Date();
 
 	const keys = await ctx.bucket.ISSUANCE_KEYS.list({ shouldUseCache: false });
 
+	if (keys.objects.length === 0) {
+		return new Response('No keys to clear', { status: 201 });
+	}
+
+	// Find the latest key based on the upload time
 	let latestKey = keys.objects[0];
-	const toDelete: Set<string> = new Set();
+	let mostRecentKeyUploadTime = new Date(latestKey.uploaded); // Assuming this is the last rotation
 
 	for (const key of keys.objects) {
-		const keyUploadTime = new Date(key.uploaded).getTime();
-		const keyExpirationTime = keyUploadTime + KEY_LIFESPAN;
-
-		if (keyExpirationTime < now) {
-			toDelete.add(key.key);
+		const keyUploadTime = new Date(key.uploaded);
+		if (keyUploadTime.getTime() > mostRecentKeyUploadTime.getTime()) {
+			mostRecentKeyUploadTime = keyUploadTime;
+			latestKey = key;
 		}
+	}
 
-		if (latestKey.uploaded < key.uploaded) {
-			latestKey = key;
+	const effectivePrevRotationTime = getPrevRotationTime(mostRecentKeyUploadTime, ctx.env);
+
+	if (effectivePrevRotationTime == null) {
+		return new Response('Failed to determine previous rotation time', { status: 500 });
+	}
+
+	const toDelete: Set<string> = new Set();
+
+	for (const key of keys.objects) {
```

Comment:
maybe a comment
```suggestion
// iterating twice over keys, because we need to know the latest upload key time to enforce key clearance
	for (const key of keys.objects) {
```